### PR TITLE
feature #82: added ability to delete comments

### LINKED ISSUE #82: Add ability to delete comments
One of the features of the admin panel will be deleting of TEXT type answers. Your task is to create an endpoint to delete a comment by question id + discipline teacher id + user id. Also add a check so that it is exactly TEXT. Return a deleted comment.

image

### PR DESCRIPTION
**Pull request by Dmutre**
- сreated deleteComment in QuestionAnswerRepository
- сreated DELETE disciplineTeachers/:disciplineTeacherId/comment endpoint
- created DeleteCommentDTO
- created DeleteCommentResponse
- unificated validateCommentBody method at QuestionAnswerService.ts
- сreated delete method to QuestionAnswerRepository
- added permission
- added Swagger documentation

closes [#82](https://github.com/fictadvisor/fictadvisor/issues/82)

### CODE REVIEW

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 1]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+  @ApiParam({
+    name: 'userId',
+    required: true,
+    description: 'User id',
+  })
+  @ApiEndpoint({
+    summary: 'Delete question answers with TEXT type (comments)',
+    permissions: PERMISSION.COMMENTS_DELETE,
+  })
+  @Delete('/:questionId/:disciplineTeacherId/:userId/comment')
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:359]: You don't have a conceptual understading what is a controller. A Controller is a bunch of endpoints which interacts with a specific entity/domain/field of project/sphere, etc. In this case, this controller intented to work with DisciplineTeacher entity. Thus, all endpoints should be connected to the discipline teacher. 

At the same time, your endpoint starts with ":questionId" param. It's not correct. It should start with ":disciplineTeacher". You can see the example at the endpoint above that edits comment. You need to have "/:disciplineTeacherId/comment" path and then use body object to define what comment should be deleted

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 2]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+  deleteComment(
+    @Param('questionId', QuestionByIdPipe) questionId: string,
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:361]: Not needed (see comment above)

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 3]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+    @Param('disciplineTeacherId', DisciplineTeacherByIdPipe) disciplineTeacherId: string,
+    @Param('userId', UserByIdPipe) userId: string
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:363]: Not needed (see comment above)

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 4]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+  ) {
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:364]: Need to add body object with information about comment (questionId, etc) for deletion

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 5]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
+  @ApiParam({
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:340]: Not needed (see comment above)

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 6]
```diff
@@ -307,4 +309,59 @@ export class DisciplineTeacherController {
...
+    name: 'questionId',
+    required: true,
+    description: 'Question Id',
+  })
+  @ApiParam({
+    name: 'disciplineTeacherId',
+    required: true,
+    description: 'Discipline teacher id',
+  })
+  @ApiParam({
```
Comment (hoshion) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:350]: Not needed (see comment above)

**Review by hoshion — CHANGES_REQUESTED** [review]

**Review by hoshion — APPROVED** [review]

**Review by hoshion — APPROVED** [review]

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 7]
```diff
@@ -307,4 +310,52 @@ export class DisciplineTeacherController {
...
+      Question has wrong type
+      
+    InvalidBodyException:
+      UserId should not be empty
+      QuestionId should not be empty`,
+  })
+  @ApiForbiddenResponse({
+    description: `
+      NoPermissionException: You do not have permission to perform this action
+    `,
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:333]: extra carriage return

File: fictadvisor-api/src/v2/api/dtos/DeleteCommentDTO.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+import { validationOptionsMsg } from '../../utils/GLOBALS';
+
+export class DeleteCommentDTO {
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/dtos/DeleteCommentDTO.ts:5]: Also inherit UpdateCommentDTO from DeleteCommentDTO

File: fictadvisor-api/src/v2/api/services/DisciplineTeacherService.ts [hunk 9]
```diff
@@ -371,4 +372,33 @@ export class DisciplineTeacherService {
...
+  async deleteComment(disciplineTeacherId: string, body: DeleteCommentDTO) {
+    await this.validateDeleteCommentBody(body);
+    this.questionAnswerRepository.delete({
+      where: {
+        disciplineTeacherId_questionId_userId: {
+          disciplineTeacherId, 
+          questionId: body.questionId, 
+          userId: body.userId
+        }
+      }
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/services/DisciplineTeacherService.ts:401]: use linter

File: fictadvisor-api/src/v2/database/repositories/QuestionAnswerRepository.ts [hunk 10]
```diff
@@ -52,4 +52,10 @@ export class QuestionAnswerRepository {
       include: this.include,
     });
   }
+
+  delete(args: Prisma.QuestionAnswerDeleteArgs) {
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/database/repositories/QuestionAnswerRepository.ts:56]: linter

File: fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts [hunk 11]
```diff
@@ -307,4 +310,52 @@ export class DisciplineTeacherController {
     const updatedComment = await this.disciplineTeacherService.updateComment(disciplineTeacherId, body);
     return this.questionMapper.getUpdatedComment(updatedComment);
   }
+
+  @ApiBearerAuth()
+  @ApiOkResponse()
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/controllers/DisciplineTeacherController.ts:315]: add response (deleted comment)

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**